### PR TITLE
Fix bugs and undefined behaviors

### PR DESCRIPTION
**Pull request by mrhhsg**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by Thearas** [comment]
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

File: be/src/pipeline/exec/partitioned_aggregation_source_operator.cpp [hunk 1]
```diff
@@ -257,7 +260,9 @@ Status PartitionedAggLocalState::recover_blocks_from_disk(RuntimeState* state, b
         size_t accumulated_blocks_size = 0;
         while (!state->is_cancelled() && !has_agg_data &&
                !_shared_state->spill_partitions.empty()) {
-            for (auto& stream : _shared_state->spill_partitions[0]->spill_streams_) {
+            while (!_shared_state->spill_partitions[0]->spill_streams_.empty() &&
+                   !state->is_cancelled()) {
+                auto& stream = _shared_state->spill_partitions[0]->spill_streams_[0];
```
Comment (yiguolei) [inline be/src/pipeline/exec/partitioned_aggregation_source_operator.cpp:265]: why change this logic？

File: be/src/pipeline/exec/partitioned_hash_join_probe_operator.cpp [hunk 2]
```diff
@@ -407,20 +407,21 @@ std::string PartitionedHashJoinProbeLocalState::debug_string(int indentation_lev
     bool need_more_input_data;
     if (_shared_state->need_to_spill) {
         need_more_input_data = !_child_eos;
-    } else if (_runtime_state) {
-        need_more_input_data = p._inner_probe_operator->need_more_input_data(_runtime_state.get());
+    } else if (_shared_state->inner_runtime_state) {
+        need_more_input_data = p._inner_probe_operator->need_more_input_data(
+                _shared_state->inner_runtime_state.get());
```
Comment (yiguolei) [inline be/src/pipeline/exec/partitioned_hash_join_probe_operator.cpp:412]: inner runtime state和  runtime_state 应该是一样的，如果不一样，逻辑上会有问题。

File: be/src/pipeline/exec/partitioned_hash_join_probe_operator.h [hunk 3]
```diff
@@ -81,7 +81,6 @@ class PartitionedHashJoinProbeLocalState final
     std::vector<vectorized::SpillStreamSPtr> _probe_spilling_streams;
 
     std::unique_ptr<vectorized::PartitionerBase> _partitioner;
-    std::unique_ptr<RuntimeState> _runtime_state;
```
Comment (yiguolei) [inline be/src/pipeline/exec/partitioned_hash_join_probe_operator.h:84]: partitioned xx operator 以后就不需要runtime state了吗？

File: be/src/vec/core/block.cpp [hunk 4]
```diff
@@ -734,7 +734,8 @@ void Block::clear_column_data(int64_t column_size) noexcept {
     }
     for (auto& d : data) {
         if (d.column) {
-            DCHECK_EQ(d.column->use_count(), 1) << " " << print_use_count();
+            // Temporarily disable reference count check because a column might be referenced multiple times within a block.
```
Comment (yiguolei) [inline be/src/vec/core/block.cpp:737]: add more comment  select c,c from table

File: be/src/vec/common/columns_hashing.h [hunk 5]
```diff
@@ -152,8 +152,12 @@ struct HashMethodSingleLowNullableColumn : public SingleColumnMethod {
 
     template <typename Data, typename Key>
     ALWAYS_INLINE FindResult find_key_with_hash(Data& data, size_t i, Key key, size_t hash_value) {
-        if (key_column->is_null_at(i) && data.has_null_key_data()) {
-            return FindResult {&data.template get_null_key_data<Mapped>(), true};
+        if (key_column->is_null_at(i)) {
+            if (data.has_null_key_data()) {
+                return FindResult {&data.template get_null_key_data<Mapped>(), true};
```
Comment (yiguolei) [inline be/src/vec/common/columns_hashing.h:157]: 之前是有什么bug 吗？

**Pull request merged**